salt/transport/ipc.py

```diff
@@ -700,11 +700,12 @@ def _read_async(self, callback):
                 for framed_msg in self.unpacker:
                     body = framed_msg['body']
                     self.io_loop.spawn_callback(callback, body)
-            except tornado.iostream.StreamClosedError:
+            except tornado.iostream.StreamClosedError as exc:
                 log.trace('Subscriber disconnected from IPC {0}'.format(self.socket_path))
-                break
+                raise exc
```

Comment:
The problem with this is that this will always be raised when closing. I would only throw if `self._closing` is False.